src/decompression.jl

```diff
@@ -514,16 +514,29 @@ end
 
 ## TreeSetColoringResult
 
+function compute_tree_value(is_star::Val{false}, B::AbstractMatrix, i::Integer, j::Integer, color::AbstractVector{<:Integer}, buffer::AbstractVector{<:Real})
+    # The tree is not a star
+    val = B[i, color[j]] - buffer[i]
+    buffer[j] = buffer[j] + val
+    return val
+end
+
+function compute_tree_value(is_star::Val{true}, B::AbstractMatrix, i::Integer, j::Integer, color::AbstractVector{<:Integer}, buffer::AbstractVector{<:Real})
+    # The tree is a star (trivial or non-trivial)
+    val = B[i, color[j]]
+    return val
+end
+
 function decompress!(
     A::AbstractMatrix, B::AbstractMatrix, result::TreeSetColoringResult, uplo::Symbol=:F
 )
-    (; ag, color, reverse_bfs_orders, buffer) = result
+    (; ag, color, reverse_bfs_orders, is_star, buffer) = result
     (; S) = ag
     uplo == :F && check_same_pattern(A, S)
     R = eltype(A)
     fill!(A, zero(R))
 
-    if eltype(buffer) == R
+    if eltype(buffer) == R || isempty(buffer)
```

Comment:
Emptiness can't be inferred statically, as a result there may be a type instability here. I suggest we get rid of this small buffer optimization
```suggestion
    if eltype(buffer) == R
```

Reply:
The idea being that currently, the type of `buffer_right_type` is not statically known, which can be bad